website/addons/box/static/boxNodeConfig.js

```diff
@@ -10,346 +10,142 @@ var $ = require('jquery');
-    self.messageClass = ko.observable('text-info');
-    // Display names
-    self.PICKER = 'picker';
-    // Current folder display
-    self.currentDisplay = ko.observable(null);
-    // CSS selector for the folder picker div
-    self.folderPicker = folderPicker;
-    // Currently selected folder, an Object of the form {name: ..., path: ...}
-    self.selected = ko.observable(null);
-    self.loading = ko.observable(false);
-    // Whether the box folders have been loaded from the server/Box API
-    self.loadedFolders = ko.observable(false);
-    self.loadedSettings = ko.observable(false);
 
-    /**
-    *  Update the view model from data returned from the server.
-    */
-    self.updateFromData = function(data) {
-        self.ownerName(data.ownerName);
-        self.nodeHasAuth(data.nodeHasAuth);
-        self.userIsOwner(data.userIsOwner);
-        self.userHasAuth(data.userHasAuth);
-        self.validCredentials(data.validCredentials);
-        // Make sure folder has name and path properties defined
-        self.folder(data.folder || {name: null, path: null, id: null});
-        self.urls(data.urls);
-    };
-
-    self.fetchFromServer = function() {
-        var request = $.ajax({
-            url: url, type: 'GET', dataType: 'json'
-        })
-        .done(function(response) {
-            self.updateFromData(response.result);
-            self.loadedSettings(true);
-            if (!self.validCredentials()){
-                if (self.userIsOwner()) {
-                    self.changeMessage('Could not retrieve Box settings at ' +
-                    'this time. The Box addon credentials may no longer be valid.' +
-                    ' Try deauthorizing and reauthorizing Box on your <a href="' +
-                        self.urls().settings + '">account settings page</a>.',
-                    'text-danger');
-                } else {
-                    self.changeMessage('Could not retrieve Box settings at ' +
-                    'this time. The Box addon credentials may no longer be valid.' +
-                    ' Contact ' + self.ownerName() + ' to verify.',
-                    'text-danger');
-                }
-            }
-        })
-        .fail(function(xhr, textStatus, error) {
-            self.changeMessage('Could not retrieve Box settings at ' +
-                'this time. Please refresh ' +
-                'the page. If the problem persists, email ' +
-                '<a href="mailto:support@osf.io">support@osf.io</a>.',
-                'text-danger');
-            Raven.captureMessage('Could not GET Box settings', {
-                url: url,
-                textStatus: textStatus,
-                error: error
-            });
+var BoxNodeConfigViewModel = oop.extend(AddonNodeConfigViewModel, {
+    constructor: function(addonName, url, selector, folderPicker) {
+        var self = this;
+        self.super.super.constructor.call(self, addonName, url, selector, folderPicker);
```

Comment:
Is there a reason you're calling `self.super.super.constructor` instead of `self.super.constructor` (the AddonNodeConfigViewModel constructor) here?